iOS Client/Reducers/MBAppReducer.swift

```diff
@@ -59,8 +65,27 @@ func devotionReducer(action: Action, state: DevotionState?) -> DevotionState {
     switch action {
     case let action as LoadedDevotions:
         nextState.devotions = action.devotions
+    case let action as SelectedDevotion:
+        nextState.selectedDevotion = action.devotion
+        if case Loaded.loaded(data: let devotions) = nextState.devotions {
+            let newDevotions: [LoadedDevotion] = devotions.map { devotion in
+                if devotion.date == nextState.selectedDevotion?.date {
+                    return LoadedDevotion(devotion: devotion.devotion, read: true)
+                } else {
+                    return devotion
+                }
+            }
+            nextState.devotions = Loaded.loaded(data: newDevotions)
+        }
+    case let action as DevotionNotification:
+        nextState.selectedDevotion = action.devotion
```

Comment:
maybe we should add `nextState.selectedDevotion.read = true` here as well ?

Reply:
Yes this is a good catch definitely need it here